support/config/deployment.go

```diff
@@ -181,65 +239,18 @@ func (c *DeploymentConfig) SetControlPlaneIsolation(hcp *hyperv1.HostedControlPl
-	if c.Replicas > 1 {
-		maxSurge := intstr.FromInt(0)
-		maxUnavailable := intstr.FromInt(1)
-		if deployment.Spec.Strategy.RollingUpdate == nil {
-			deployment.Spec.Strategy.RollingUpdate = &appsv1.RollingUpdateDeployment{}
-		}
-		deployment.Spec.Strategy.RollingUpdate.MaxSurge = &maxSurge
-		deployment.Spec.Strategy.RollingUpdate.MaxUnavailable = &maxUnavailable
-	}
-
-	// set default security context for pod
-	if c.SetDefaultSecurityContext {
-		deployment.Spec.Template.Spec.SecurityContext = &corev1.PodSecurityContext{
-			RunAsUser: pointer.Int64(DefaultSecurityContextUser),
-		}
-	}
-
-	// set managed-by label
-	if deployment.Labels == nil {
-		deployment.Labels = map[string]string{}
+// setNodeSelector sets a nodeSelector passed through the API.
+// This is useful to e.g ensure control plane pods land in management cluster Infra Nodes.
+func (c *DeploymentConfig) setNodeSelector(hcp *hyperv1.HostedControlPlane) {
+	if hcp.Spec.NodeSelector == nil {
+		return
 	}
-	deployment.Labels[ManagedByLabel] = "control-plane-operator"
-
-	c.Scheduling.ApplyTo(&deployment.Spec.Template.Spec)
-	c.AdditionalLabels.ApplyTo(&deployment.Spec.Template.ObjectMeta)
-	c.SecurityContexts.ApplyTo(&deployment.Spec.Template.Spec)
-	c.Resources.ApplyTo(&deployment.Spec.Template.Spec)
-	c.LivenessProbes.ApplyTo(&deployment.Spec.Template.Spec)
-	c.ReadinessProbes.ApplyTo(&deployment.Spec.Template.Spec)
-	c.Resources.ApplyTo(&deployment.Spec.Template.Spec)
-	c.AdditionalAnnotations.ApplyTo(&deployment.Spec.Template.ObjectMeta)
+	c.Scheduling.NodeSelector = hcp.Spec.NodeSelector
 }
 
-func (c *DeploymentConfig) ApplyToDaemonSet(daemonset *appsv1.DaemonSet) {
-	// replicas is not used for DaemonSets
-	c.Scheduling.ApplyTo(&daemonset.Spec.Template.Spec)
-	c.AdditionalLabels.ApplyTo(&daemonset.Spec.Template.ObjectMeta)
-	c.SecurityContexts.ApplyTo(&daemonset.Spec.Template.Spec)
-	c.Resources.ApplyTo(&daemonset.Spec.Template.Spec)
-	c.LivenessProbes.ApplyTo(&daemonset.Spec.Template.Spec)
-	c.ReadinessProbes.ApplyTo(&daemonset.Spec.Template.Spec)
-	c.Resources.ApplyTo(&daemonset.Spec.Template.Spec)
-	c.AdditionalAnnotations.ApplyTo(&daemonset.Spec.Template.ObjectMeta)
-}
-
-func (c *DeploymentConfig) ApplyToStatefulSet(sts *appsv1.StatefulSet) {
-	sts.Spec.Replicas = pointer.Int32Ptr(int32(c.Replicas))
-	c.Scheduling.ApplyTo(&sts.Spec.Template.Spec)
-	c.AdditionalLabels.ApplyTo(&sts.Spec.Template.ObjectMeta)
-	c.SecurityContexts.ApplyTo(&sts.Spec.Template.Spec)
-	c.Resources.ApplyTo(&sts.Spec.Template.Spec)
-	c.LivenessProbes.ApplyTo(&sts.Spec.Template.Spec)
-	c.ReadinessProbes.ApplyTo(&sts.Spec.Template.Spec)
-	c.Resources.ApplyTo(&sts.Spec.Template.Spec)
-	c.AdditionalAnnotations.ApplyTo(&sts.Spec.Template.ObjectMeta)
+func (c *DeploymentConfig) SetLocation(hcp *hyperv1.HostedControlPlane, multiZoneSpreadLabels map[string]string) {
+	c.setNodeSelector(hcp)
+	c.setControlPlaneIsolation(hcp)
+	c.setColocation(hcp)
+	c.setMultizoneSpread(multiZoneSpreadLabels)
```

Comment:
This should only be called if the deployment has replicas > 1, otherwise it prevents single replica deployment rollouts.